### PR TITLE
Release notes and Migration guide for Hostname v2

### LINKED ISSUE #27730: Release notes and Migration guide for Hostname v2
### Description

Add a chapter to the [upgrading guide](https://github.com/keycloak/keycloak/blob/main/docs/documentation/upgrading/topics/keycloak/changes-25_0_0.adoc) covering migration for Hostname v2. It should show some examples of old vs. new config (a table?) both for server and the operator.

This should be also linked to from the release notes.

### Discussion

_No response_

### Motivation

_No response_

### Details

_No response_

### PR DESCRIPTION
**Pull request by mabartos**
Closes #27730
Blocked-by: https://github.com/keycloak/keycloak/pull/28123

Not sure whether it should be more comprehensive, but I think it's sufficient as the guide contains all the necessary information.

### CODE REVIEW

File: docs/documentation/upgrading/topics/changes/changes-25_0_0.adoc [hunk 1]
```diff
@@ -1,3 +1,37 @@
...
+As you can see, the `*-url` suffixes were removed for `hostname`, and `hostname-admin` options.
+These options accept both hostname, and URL now.
+
+Additionally, there is no way how to separately set `path`, or `port`.
+You can achieve it by providing the full URL for the `hostname` option.
+
+NOTE: In order to use the old hostname options, turn on the feature `hostname:v1`, e.g. `features=hostname:v1`.
+Be aware, that either `hostname:v1`, or `hostname:v2` can be enabled, not both.
+
+For more details, and more comprehensive scenarios, see https://www.keycloak.org/server/hostname2[Configuring the hostname (v2)].
```
Comment (Pepo48) [inline docs/documentation/upgrading/topics/changes/changes-25_0_0.adoc:33]:
```suggestion
For more details, and more comprehensive scenarios, see https://www.keycloak.org/server/hostname[Configuring the hostname (v2)].
```

File: docs/documentation/upgrading/topics/changes/changes-25_0_0.adoc [hunk 2]
```diff
@@ -1,3 +1,37 @@
...
+`hostname-port <port>`
+|`hostname <hostname/url>`
+|`hostname-admin <hostname>` +
+`hostname-admin-url <url>`
+|`hostname-admin <hostname/url>`
+|`hostname-strict-backchannel <true/false>`
+|`hostname-backchannel-dynamic <true/false>`
+|===
+
+As you can see, the `*-url` suffixes were removed for `hostname`, and `hostname-admin` options.
```
Comment (Pepo48) [inline docs/documentation/upgrading/topics/changes/changes-25_0_0.adoc:24]:
```suggestion
As you can see, the `*-url` suffixes were removed for `hostname` and `hostname-admin` options.
```

File: docs/documentation/upgrading/topics/changes/changes-25_0_0.adoc [hunk 3]
```diff
@@ -1,3 +1,37 @@
...
+These options accept both hostname, and URL now.
+
+Additionally, there is no way how to separately set `path`, or `port`.
```
Comment (Pepo48) [inline docs/documentation/upgrading/topics/changes/changes-25_0_0.adoc:27]:
```suggestion
Additionally, there is no way how to separately set `path` or `port`.
```

File: docs/documentation/upgrading/topics/changes/changes-25_0_0.adoc [hunk 4]
```diff
@@ -1,3 +1,37 @@
...
+You can achieve it by providing the full URL for the `hostname` option.
+
+NOTE: In order to use the old hostname options, turn on the feature `hostname:v1`, e.g. `features=hostname:v1`.
+Be aware, that either `hostname:v1`, or `hostname:v2` can be enabled, not both.
```
Comment (Pepo48) [inline docs/documentation/upgrading/topics/changes/changes-25_0_0.adoc:31]:
```suggestion
Be aware, that either `hostname:v1` or `hostname:v2` can be enabled, not both.
```

**Comment by mabartos** [comment]
@vmuzikar Could you please point me to what examples would be most suitable to mention in the migration guide?

**Comment by vmuzikar** [comment]
> @vmuzikar Could you please point me to what examples would be most suitable to mention in the migration guide?

@mabartos I'd probably choose some more complex scenarios that uses as many old options as possible (admin-url, -path, -port, ...). We should also emphasise we no longer enforce HTTPS unless it's part of hostname-url, and that we don't assume any ports when proxy is used. See the [old docs](https://github.com/keycloak/keycloak/blob/6d74e6b2895d8b0820d910a818bef8f67d0df129/docs/guides/server/hostname.adoc?plain=1#L43-L44).

**Review by vmuzikar — CHANGES_REQUESTED** [review]
@mabartos Thank you for the update. Added some comments.

**Review by vmuzikar — DISMISSED** [review]
LGTM, thanks!

File: docs/documentation/upgrading/topics/changes/changes-25_0_0.adoc [hunk 5]
```diff
@@ -1,3 +1,85 @@
...
+.Simplified notation
+[source,bash]
+----
+# Hostname v1
+bin/kc.[sh|bat] start --hostname=mykeycloak.org --https-port=8543 --hostname-path=/auth --hostname-strict-https=true
+
+# Hostname v2
+bin/kc.[sh|bat] start --hostname=https://mykeycloak.org:8543/auth
+----
+As the `hostname` now accepts hostname and URL, we can simplify the hostname settings directly by it.
```
Comment (Pepo48) [inline docs/documentation/upgrading/topics/changes/changes-25_0_0.adoc:62]: Here I would get rid of "we" pronoun to redirect the narrative more towards "them". I assume you wanted to say that the setup process is now a lot simpler. If so, I would rephrase it a bit to make the sentence a bit more clear, something like: 
```suggestion
As you can see in the example, all the parts of a URL can be now specified via single `hostname` option, which simplifies the hostname setup process.
```

File: docs/documentation/upgrading/topics/changes/changes-25_0_0.adoc [hunk 6]
```diff
@@ -1,3 +1,85 @@
...
+.Backchannel setting
+[source,bash]
+----
+# Hostname v1
+bin/kc.[sh|bat] start --hostname=mykeycloak.org --hostname-strict-backchannel=true
+
+# Hostname v2
+bin/kc.[sh|bat] start --hostname=mykeycloak.org --hostname-backchannel-dynamic=false
+----
+In order to use the same URL for backend as for frontend endpoints, we need to be aware of a different behavior between hostname versions.
```
Comment (Pepo48) [inline docs/documentation/upgrading/topics/changes/changes-25_0_0.adoc:74]:
```suggestion
Be aware that there is a change in behavior if the same URL is to be used for both backend and frontend endpoints.
```

**Review by Pepo48 — DISMISSED** [review]
@mabartos nice work, thank you!

I have just small nitpicks to consider, related to clarification of two sentences. I'll leave it up to you, whether you want to include it or not.

Other than that LGTM, therefore I'm approving the PR.

**Comment by mabartos** [comment]
@Pepo48 make sense to me. Thanks!

@Pepo48 @vmuzikar Ready for another round of reviews.

**Review by Pepo48 — APPROVED** [review]

**Review by vmuzikar — APPROVED** [review]
